pkg/notifications/fetch.go

```diff
@@ -0,0 +1,159 @@
+package notifications
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/icinga/icinga-go-library/backoff"
+	"github.com/icinga/icinga-go-library/retry"
+	"github.com/icinga/icinga-go-library/types"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+	"time"
+)
+
+// fetchHostServiceFromRedis retrieves the host and service names from Redis.
+//
+// If serviceId is nil, only the host name is fetched. Otherwise, both host and service name is fetched.
+func (client *Client) fetchHostServiceFromRedis(
+	ctx context.Context,
+	hostId, serviceId types.Binary,
+) (hostName string, serviceName string, err error) {
+	getNameFromRedis := func(ctx context.Context, typ, id string) (string, error) {
+		key := "icinga:" + typ
+
+		var data string
+		if err := retry.WithBackoff(
+			ctx,
+			func(ctx context.Context) (err error) {
+				data, err = client.redisClient.HGet(ctx, key, id).Result()
+				return
+			},
+			retry.Retryable,
+			backoff.DefaultBackoff,
+			retry.Settings{},
+		); err != nil {
```

Comment:
IMHO that's an example where it definitely doesn't make sense to initialize the variable within the `if` statement.
```suggestion
		err := retry.WithBackoff(
			ctx,
			func(ctx context.Context) (err error) {
				data, err = client.redisClient.HGet(ctx, key, id).Result()
				return
			},
			retry.Retryable,
			backoff.DefaultBackoff,
			retry.Settings{})
		if err != nil {
```